apps/faucet/src/utils/api.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  ChallengeResponse,
+  Data,
+  SettingsResponse,
+  TransferResponse,
+} from "./types";
+
+enum Endpoint {
+  Settings = "/setting",
+  Challenge = "/challenge",
+  Transfer = "",
+}
+
+export class API {
+  constructor(protected readonly url: string) {}
+
+  /**
+   * Wrapper for fetch requests to handle ReadableStream response when errors are received from API
+   *
+   * @param {string} endpoint
+   * @param {RequestInit} options
+   *
+   * @returns Object
+   */
+  async request<T = unknown>(
+    endpoint: string,
+    options: RequestInit = { method: "GET" }
+  ): Promise<T> {
+    return (await fetch(new URL(`${this.url}${endpoint}`), {
+      ...options,
+    })
+      .then((response) => {
+        if (response.ok) {
+          return response.json();
+        }
+        const reader = response?.body?.getReader();
+        return reader
+          ?.read()
+          .then((data) =>
+            Promise.reject(JSON.parse(new TextDecoder().decode(data.value)))
+          );
+      })
+      .catch((e) => {
+        console.error(e);
+        return Promise.reject(e);
+      })) as T;
```

Comment:
I think this will compile without `as T`.

Reply:
Thanks! Will remove it!